Issue 40: Add example test data

athowes opened the pull request:
<!-- Thanks for opening this pull request! Below we have provided a suggested template for PRs to this repository and a checklist to complete before opening a PR -->
 
## Description

This PR closes #40.

I have some remaining uncertainty about:

1. How accessible this data is (internal or external to the package)
2. Should the data be documented?
3. Does the `set.seed` work as intended? What about that `simulate_gillespie()` has a `set.seed` argument?

## Checklist

- [ ] My PR is based on a package issue and I have explicitly linked it.
- [ ] I have included the target issue or issues in the PR title in the for Issue(s) *issue-numbers*: PR title
- [ ] I have read the [contribution guidelines](https://github.com/epinowcast/.github/blob/main/CONTRIBUTING.md).
- [ ] I have tested my changes locally.
- [ ] I have added or updated unit tests where necessary.
- [ ] I have updated the documentation if required.
- [ ] My code follows the established coding standards.
- [ ] I have added a news item linked to this PR.
- [ ] I have reviewed CI checks for this PR and addressed them as far as I am able.

<!-- Thanks again for this PR - @epinowcast dev team -->

seabbs reviewed (commented):
I think this is fine and a reasonable thing to do but note that CRAN can get funny about including data just for examples. As flagged in the issue we might want to consider the alternative approach `epinowcast` users or provide some pushback as to why we should take this approach

athowes commented:
I've made the test data internal.

As for the seed issue, I've rerun it two times and the results are the same, so seems fine.

athowes commented:
I'm pretty indifferent about it being in `data/` and accessible or this internal one. I don't know if there are other options.

athowes commented:
Have reverted back to data being a part of the package, and added documentation. Ready for review @seabbs.

athowes commented:
From https://r-pkgs.org/data.html:

> If you want to store R objects and make them available to the user, put them in data/. This is the best place to put example datasets. All the concrete examples above for data in a package and data as a package use this mechanism. See section [Section 7.1](https://r-pkgs.org/data.html#sec-data-data).

Not perfect fit but arguably the test data could be helpful to a user.

> If you want to store R objects for your own use as a developer, put them in R/sysdata.rda. This is the best place to put internal data that your functions need. See section [Section 7.2](https://r-pkgs.org/data.html#sec-data-sysdata).

The test data are for own use as a developer. It's not "internal data that functions need" though.

> If you want to store data in some raw, non-R-specific form and make it available to the user, put it in inst/extdata/. For example, readr and readxl each use this mechanism to provide a collection of delimited files and Excel workbooks, respectively. See section [Section 7.3](https://r-pkgs.org/data.html#sec-data-extdata).

No.

> If you want to store dynamic data that reflects the internal state of your package within a single R session, use an environment. This technique is not as common or well-known as those above, but can be very useful in specific situations. See section [Section 7.4](https://r-pkgs.org/data.html#sec-data-state).

No.

> If you want to store data persistently across R sessions, such as configuration or user-specific data, use one of the officially sanctioned locations. See section [Section 7.5](https://r-pkgs.org/data.html#sec-data-persistent).

No.

seabbs commented:
@athowes and I had a f2f chat and concluded we should use simulated data in the tests via `test/testthat/setup.R` and include the PNAS data for examples / real-world context (with lots of documentation to point back at the original source).

athowes commented:
* Added simulated data in `test/testthat/setup.R`
* Added back ebola data. Got `.xlsx` direct download from paper in `data-raw`. Used `data-raw/process_raw_data.R` to lightly process and save to `.rda`. Documented data in `R/data.R`

seabbs commented on R/data.R:
```diff
@@ -0,0 +1,18 @@
+#' }
+#' @family data
+#' @source <https://www.pnas.org/doi/full/10.1073/pnas.1518587113>
+"ebola"
```
I think we might want to be a little more verbose here i.e `sierra_leone_ebola_outbreak_data` or similar?

seabbs replied:
How do people know that just from looking? It could be a function

seabbs replied:
I very much don't think it should just be ebola - location context is important

athowes replied:
Yes that's fair r.e. it could be a function

seabbs commented on R/data.R:
```diff
@@ -0,0 +1,18 @@
+#' Ebola linelist data from Fang et al. (2016)
+#'
+#' Linelist data for the Ebola virus collected in Sierra Leone.
```
Here might be a good opportunity to ask  people explicitly to cite

athowes replied:
Yes can do. Is there a nice way to include bibtex in `roxygen2` or I can just write "please cite"

seabbs replied:
you should be able to use a @reference macro I think but not sure what it gets you vs @source. I am really just suggesting saying please cite

seabbs commented on R/data.R:
```diff
@@ -0,0 +1,18 @@
+#' Ebola linelist data from Fang et al. (2016)
```
roxygen needs to be generated

athowes replied:
I am a little confused in general about when to generate `roxygen2`. I suppose after each change which alters it is reasonable.

seabbs replied:
yes. I agree its pretty boring. An approach I like the idea of but haven't done is to leave this as a CI job for main. The other option is to pin it into `pre-commit` so it always get done for you (we aren' using pre-commit here but maybe wee should)

seabbs commented on R/data.R:
```diff
@@ -0,0 +1,18 @@
+#' }
+#' @family data
+#' @source <https://www.pnas.org/doi/full/10.1073/pnas.1518587113>
+"ebola"
```
missing end of line

seabbs commented on data-raw/process_raw_data.R:
```diff
@@ -0,0 +1,3 @@
+ebola <- readxl::read_xlsx("data-raw/pnas.1518587113.sd02.xlsx")
```
These are now development dependencies and so potentially should go in the suggests section of the description

seabbs commented on data-raw/process_raw_data.R:
```diff
@@ -0,0 +1,3 @@
+ebola <- readxl::read_xlsx("data-raw/pnas.1518587113.sd02.xlsx")
+ebola <- janitor::clean_names(ebola)
```
good idea to use `janitor`

seabbs reviewed (changes requested):
This all looks good. A few minor comments

athowes commented:
Have fixed these comments I think.

seabbs reviewed (approved):
All good to go now except the minor linting issues

seabbs commented on data-raw/process_raw_data.R:
```diff
@@ -0,0 +1,6 @@
+sierra_leone_ebola_outbreak_data <- readxl::read_xlsx(
```
```suggestion
sierra_leone_ebola_outbreak_data <-
	readxl::read_xlsx(
```

athowes replied:
What's the style guide here?

seabbs replied:
I mean this isn't ideal but is what `lintr` will suggest with these very long names. I agree with your name change so can resolve this.

athowes commented:
Failing lint due to "Variable and function names should not be longer than 30 characters."

`sierra_leone_ebola_outbreak_data` change to `sierra_leone_ebola_data` perhaps.
